corehq/apps/users/forms.py

```diff
@@ -869,6 +869,15 @@ def clean_assigned_locations(self):
 
         return [location.location_id for location in locations]
 
+    def clean(self):
+        cleaned_data = super(CommtrackUserForm, self).clean()
+
+        primary_location_id = cleaned_data['primary_location']
+        assigned_location_ids = cleaned_data['assigned_locations']
+        if primary_location_id:
+            if primary_location_id not in assigned_location_ids:
+                raise ValidationError(_("Only one of the user's locations can be a primary location"))
```

Comment:
Since this validation is related to two fields, the error can't be captured in a field specific clean method, that's why I had to put this here